pkg/channel/testing/channel_mock.go

```diff
@@ -82,9 +84,11 @@ func (_m *MockAppChannel) Init() {
 
 // InvokeMethod provides a mock function with given fields: ctx, req
 func (_m *MockAppChannel) InvokeMethod(ctx context.Context, req *v1.InvokeMethodRequest) (*v1.InvokeMethodResponse, error) {
+	_m.mutex.Lock()
 	if _m.requestsReceived != nil {
 		_m.requestsReceived[req.Message().Method] = req
 	}
+	_m.mutex.Unlock()
 	ret := _m.Called(ctx, req)
 
 	var r0 *v1.InvokeMethodResponse
```

Comment:
(For line 114) shoulodn't there be a lock on `GetInvokeRequest` too?

Reply:
Agreed, incorporated.